scripts/search-res-xml.py

```diff
@@ -116,14 +181,52 @@ def match_frag_score(traverse_node, frag_node: Element):
         return 0.5
     return 0.2
 
+def d(o):
```

Comment:
This is very surprising -- what is it for? What do you mean by circular reference?

Reply:
From what I recall: the JSON object contains the reference to the parent node, and the parent node contains a reference to its children. When we print the children, its parent will be printed, and when the parent is printed, it will print the children. Therefore the object cannot be converted to a dict directly. 
I added this condition so that the everything cannot be made to a dict is not printed. (In this case, it is because of the circular reference)

Reply:
Oh I see.